src/probnum/diffeq/perturbedsolvers/_perturbation_functions.py

```diff
@@ -0,0 +1,58 @@
+"""perturbation functions to perturb the stepsize.
+
+References
+----------
+.. [1] https://arxiv.org/abs/1801.01340
+"""
+import numpy as np
+import scipy
+
+
+def perturb_uniform(step, solver_order, noise_scale, random_state=None, size=()):
+    """Perturb the step with uniformly distributed noise scaled by noise-scale [1]_.
+    proposed by Abdulle and Garegnani(2020)
+
+    Parameters
+    ----------
+    step : float
+        unperturbed step propesed by the steprule
+    solver_order : int
+        order of the solver
+    noise_scale : float
+        scales the perturbation
+    seed : int
+        seed for pseudo-random number generator.
+    """
+    scipy.stats.uniform.rvs(random_state=random_state)
```

Comment:
What I meant with me comment in the previous review, the code in this function should be along the lines of
1) `if step >= 1.: raise ValueError("Stepsize too large (>= 1)")`, which raises the error correctly
2) `uniform_rv_samples = scipy.stats.uniform.rvs(random_state=random_state, size=size)`  which computs standard (0,1) uniform samples
3) `scaled_uniform_rv_samples = left_boundary + (right_boundary - left_boundary)*uniform_rv_samples`, where `left_boundary=step - noise_scale * step...` and `right_boundary=step + noise_scale...`. This moves the uniform samples to the right domain

This is different to your implementation in that it properly handles `size`, `random_state`, and `step >= 1.`. Sorry if my comment in the previous review was not sufficiently accurate, I should've been more clear with the `scipy.stats.uniform` mention. If there are any questions, please let me know, okay? :)

Reply:
it might also be good to compute the shift `noise_scale * step**(solver_order + 0.5)` only once (for simplicity and symmetry, and to a probably neglectible amount: speed).